### ISSUE TITLE
feat: castor record --annotate — label video segments with actions

### ISSUE THREAD

**Issue by craigm26** [craigm26/OpenCastor#233]
## Summary
Extend the existing video recorder (`castor/recorder.py`) to write action annotations alongside video frames, producing a labelled dataset for robot learning fine-tuning.

## Implementation
- `VideoRecorder.annotate(action: dict)` — writes frame timestamp + action to a sidecar JSON-L file
- `GET /api/recording/{id}/annotations` — returns annotation JSON-L
- `castor record --annotate` CLI flag enables annotation mode
- Export: `castor export-finetune --from-recording {id}` converts annotated recording to Alpaca/ShareGPT format
- Dashboard: timeline scrubber showing action labels over video playback

## API additions
```
GET  /api/recording/{id}/annotations   → [{ts_ms, action, frame_idx}]
POST /api/recording/{id}/annotate      → {ts_ms, action}   (manual post-hoc annotation)
```

## Acceptance criteria
- [ ] Action annotations written to sidecar JSON-L during `castor run`
- [ ] Annotated recording exports to Alpaca fine-tune format
- [ ] Dashboard playback shows action overlays
- [ ] Tests cover annotation write/read and export
